### PR TITLE
Adds %Ash.NotSelected{} to ash 3 (!634)

### PR DESCRIPTION
**Pull request by ghalestrilo**
Introduces an [`%Ash.NotSelected{}`](https://github.com/ash-project/ash/issues/634) struct to ash >3.0.0, replacing `nil` in unselected fields during a read operation.

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [x] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by zachdaniel** [comment]
Nice! What we should also do is update the GitHub CI file to run tests against ash_three as well, so we can see what breaks (if anything).

File: lib/ash/actions/helpers.ex [hunk 1]
```diff
@@ -653,7 +654,14 @@ defmodule Ash.Actions.Helpers do
       end
     end)
     |> Enum.reduce(result, fn key, record ->
-      Map.put(record, key, nil)
+      default_field_value =
+        if Ash.Flags.ash_three?() do
+          %Ash.NotSelected{}
```
Comment (zachdaniel) [inline lib/ash/actions/helpers.ex:659]: This should include the field that is not selected.

**Comment by ghalestrilo** [comment]
I noticed that the ash three tests are missing in CI!  But they'll require a dedicated build step as well 😬 
Does it makes more sense to be a separate MR?

**Comment by zachdaniel** [comment]
Yeah, I think a separate PR is good :)

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**